### PR TITLE
Link supports client-side navigation

### LINKED ISSUE #822: Review Link implementation for client side navigation

### PR DESCRIPTION
**Pull request by marcialfps**
### New features in DxcLink
- Removed `text` prop.
- Added `children` as prop. Can be a `string` or `ReactNode`.
- Updated Link tests and stories.
- Updated Link documentation and changed some examples.

### Other changes
- Updated `docs` and `app` with DxcLink changes.
- Use of React Router combined with DxcLink in `docs`.

Closes #822

### CODE REVIEW

File: docs/src/pages/components/cdk-components/link/api.jsx [hunk 1]
```diff
@@ -63,6 +58,16 @@ const linkPropsTable = () => {
           be called when the user clicks the link.
         </td>
       </tr>
+      <tr>
+        <td>children: string | node</td>
+        <td></td>
+        <td>
+          Content to display in the link. To use the link with other libraries
```
Comment (GomezIvann) [inline docs/src/pages/components/cdk-components/link/api.jsx:65]: I think this description is a bit verbose, maybe reformulate it like: "Content of the link. To use the component with other libraries (such as React Router or Next.js Link) pass the custom link as a child. The component will apply the styles to the custom link.", what do you think?

File: docs/src/pages/overview/Overview.jsx [hunk 2]
```diff
@@ -62,8 +59,9 @@ function UseComponents() {
           href="https://developer.dxc.com/design/components"
```
[the remark this answers is not in the document]
Comment (marcialfps) [reply docs/src/pages/overview/Overview.jsx:60]: Fixed. Changed to `/components`.

File: docs/src/pages/components/cdk-components/link/api.jsx (hunk 1 above)
Comment (marcialfps) [reply docs/src/pages/components/cdk-components/link/api.jsx]: Yes, now is better 👍🏼

File: lib/src/link/Link.tsx [hunk 3]
```diff
@@ -78,6 +78,7 @@ const LinkContainer = styled.div<LinkContainerProps>`
     padding: 0;
     cursor: pointer;
     font-family: inherit;
+    font-size: inherit;
```
Comment (GomezIvann) [inline lib/src/link/Link.tsx:81]: Shouldn't this rule have the token value? Same question goes for font-family, font-style, font-weight, etc.

Comment (GomezIvann) [reply lib/src/link/Link.tsx:81]: Currently they are at `LinkText` but they won't affect every version of the link, no? 🤔

Comment (marcialfps) [reply lib/src/link/Link.tsx]: Done! ✌🏼

**Review by GomezIvann — APPROVED** [review]

**Pull request closed without merging**